sdk/servicebus/azure-servicebus/azure/servicebus/exceptions.py

```diff
@@ -228,108 +193,245 @@ class ServiceBusError(AzureError):
+            message,
+            retryable=False,
+            shutdown_handler=False,
+            **kwargs
+        )
 
 
-class SessionLockExpired(ServiceBusError):
-    """The lock on the session has expired.
+class SessionLockLostError(ServiceBusError):
+    """The lock on the session has expired. Callers should request the session again.
 
     All unsettled messages that have been received can no longer be settled.
-
     """
 
-    def __init__(self, message=None, error=None):
-        # type: (Optional[str], Optional[Exception]) -> None
-        message = message or "Session lock expired"
-        super(SessionLockExpired, self).__init__(message, error=error)
+    def __init__(self, **kwargs):
+        # type: (Any) -> None
+        message = kwargs.pop("message", None) or\
+                  "The lock on the session has expired. Callers should request the session again."
+        super(SessionLockLostError, self).__init__(
+            message,
+            retryable=False,
+            shutdown_handler=True,
+            **kwargs
+        )
+
+
+class MessageNotFoundError(ServiceBusError):
+    """The requested message was not found."""
+    def __init__(self, **kwargs):
+        message = kwargs.pop("message", None) or "The requested message was not found."
+        super(MessageNotFoundError, self).__init__(
+            message,
+            retryable=False,
+            shutdown_handler=False,
+            **kwargs
+        )
+
+
+class MessagingEntityNotFoundError(ServiceBusError):
+    """A Service Bus resource cannot be found by the Service Bus service."""
+    def __init__(self, **kwargs):
+        message = kwargs.pop("message", None) or "A Service Bus resource cannot be found by the Service Bus service."
+        super(MessagingEntityNotFoundError, self).__init__(
+            message,
+            retryable=False,
+            shutdown_handler=False,
+            **kwargs
+        )
+
+
+class MessagingEntityDisabledError(ServiceBusError):
+    """The Messaging Entity is disabled. Enable the entity again using Portal."""
+    def __init__(self, **kwargs):
+        message = kwargs.pop("message", None) or \
+                  "The Messaging Entity is disabled. Enable the entity again using Portal."
+        super(MessagingEntityDisabledError, self).__init__(
+            message,
+            retryable=True,
+            shutdown_handler=True,
+            **kwargs
+        )
+
+
+class MessagingEntityAlreadyExistsError(ServiceBusError):
+    """An entity with the same name exists under the same namespace."""
+    def __init__(self, **kwargs):
+        message = kwargs.pop("message", None) or "An entity with the same name exists under the same namespace."
+        super(MessagingEntityAlreadyExistsError, self).__init__(
+            message,
+            retryable=False,
+            shutdown_handler=False,
+            **kwargs
+        )
+
+
+class QuotaExceededError(ServiceBusError):
```

Comment:
Is this "generic"?  I worry in an Azure situation, folks may wonder if it's an Azure quota or SB specific.  cc @annatisch

Reply:
as per oob discussion:

QuotaExceededError -> ServiceBusQuotaExceededError
ServiceBusyError -> ServiceBusServerBusyError